src/Guzzle/SubjectRewritingMiddleware.php

```diff
@@ -0,0 +1,192 @@
+                    case 'application/vnd.elife.press-package-list+json;version=1':
+                    case 'application/vnd.elife.recommendations+json;version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+
+                    case 'application/vnd.elife.article-poa+json;version=2':
+                    case 'application/vnd.elife.article-vor+json;version=2':
+                    case 'application/vnd.elife.blog-article+json;version=2':
+                        $data = $this->updateItem($data);
+                        break;
+
+                    case 'application/vnd.elife.collection+json;version=1':
+                        $data = $this->updateItem($data);
+                        $data['content'] = $this->updateItems($data['content']);
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.podcast-episode+json;version=1':
+                        $data['chapters'] = array_map(function (array $chapter) {
+                            $chapter['content'] = $this->updateItems($chapter['content'] ?? []);
+
+                            return $chapter;
+                        }, $data['chapters']);
+                        break;
+
+                    case 'application/vnd.elife.person-list+json;version=1':
+                        $data['items'] = array_map([$this, 'updatePerson'], $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.person+json;version=1':
+                        $data = $this->updatePerson($data);
+                        break;
+
+                    case 'application/vnd.elife.press-package+json;version=3':
+                        $data = $this->updateItem($data);
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.search+json;version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        foreach ($this->replacements as $replaced => $replacement) {
+                            $total = 0;
+                            foreach ($data['subjects'] as $i => $subject) {
+                                if ($replaced === $subject['id']) {
+                                    $total = $subject['results'];
+                                    unset($data['subjects'][$i]);
+                                    break;
+                                }
+                            }
+                            if ($total > 0) {
+                                foreach ($data['subjects'] as $i => $subject) {
+                                    if ($replacement['id'] === $subject['id']) {
+                                        $data['subjects'][$i]['results'] += $total;
```

Comment:
Seems to produce a different number to what's returned...

Reply:
Ah, articles with both. We can't know what the real number should be...

Reply:
Will have a look at the test cases, this is the most complex case